### PR TITLE
CAM misc tag for derecho testing, fix SILHS output, fix single-exe for PLB tests and fix HEMCO line too long

### PR DESCRIPTION
**Pull request by cacraigucar**
was addressing #879 (Add derecho support for test_driver)
was addressing #792  (changes to fix single-exe option for PLB tests) 
was addressing #855 (Bug fix for SILHS history output being garbled)
was addressing #886 (Fix HEMCO line too long)

### CODE REVIEW

**Comment by fvitt** [comment]
I believe you should update the external ccs_config tag to ccs_config_cesm0.0.75 for derecho testing

**Comment by cacraigucar** [comment]
I have a request to have an updated ccs_config tag made which allows cheyenne tests to pass for all users (so it hopefully that external will be updated).   Feel free to review the rest of this PR, knowing that the ccs_config external may be updated

**Review by jtruesdal — APPROVED** [review]
Looks good.  While reviewing I did notice that the fms library updates are not consistent with the current fv3 tag.  Nice catch.  FMS will be updated with an fv3 PR in the near future.

File: test/system/test_driver.sh [hunk 1]
```diff
@@ -273,6 +273,63 @@ EOF
...
+    # These settings are ignored on cheyenne and derecho.
+    # PE layouts come from config_pes.xml.
+    CAM_TASKS=64
+    CAM_THREADS=2
+
+    # change parallel configuration on 1 nodes using 32 tasks, 1 threads/task
+    # These settings are ignored on cheyenne and derecho.
+    # PE layouts come from config_pes.xml.
+    CAM_RESTART_TASKS=32
+    CAM_RESTART_THREADS=1
```
Comment (nusbaume) [inline test/system/test_driver.sh:306]: Why set these variables at all if they are just going to be ignored?

**Review by nusbaume — CHANGES_REQUESTED** [review]
Looks good, but I did have a few minor test_driver questions and change requests.

File: test/system/test_driver.sh [hunk 2]
```diff
@@ -273,6 +273,63 @@ EOF
...
+
+    mach_workspace="/glade/derecho/scratch"
+
+    # Check for CESM baseline directory
+    if [ -n "${BL_TESTDIR}" ] && [ ! -d "${BL_TESTDIR}" ]; then
+        echo "CESM_BASELINE ${BL_TESTDIR} not found.  Check BL_TESTDIR for correct tag name."
+        exit
```
[the remark this answers is not in the document]
Comment (fischer-ncar) [reply test/system/test_driver.sh:313]: Agreed, I'll add a non-zero exit code.

File: test/system/test_driver.sh (hunk 1 above)
Comment (fischer-ncar) [reply test/system/test_driver.sh:306]: I believe that if you force all of the tests to run on a single node, like on izumi, then these variables will probably be used.  I was just concentrating on getting the tests running on derecho and didn't wont to dig too deep into the script.

File: test/system/test_driver.sh [hunk 3]
```diff
@@ -273,6 +273,63 @@ EOF
...
+
+#-------------------------------------------
+
+cat > ${submit_script_cime} << EOF
+#!/bin/bash
+#
+#PBS -N cime-tests
+#PBS -q $CAM_BATCHQ
+#PBS -A $CAM_ACCOUNT
+#PBS -l walltime=4:00:00
```
[the remark this answers is not in the document]
Comment (fischer-ncar) [reply test/system/test_driver.sh:324]: It should be walltime.  All of the CISL examples use walltime, and ccs_config is also using walltime.

File: test/system/test_driver.sh (hunk 1 above)
Comment (nusbaume) [reply test/system/test_driver.sh:306]: Ahh, that makes sense.  I'm fine to leave this as-is then.

File: test/system/test_driver.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (nusbaume) [reply test/system/test_driver.sh:324]: Sorry, I wasn't referring to the `walltime` PBS variable, I was referring to the actual `wallclock_limit` variable set earlier in the script. In other words should this line be:

```suggestion
#PBS -l walltime=$wallclock_limit
```

[the remark this answers is not in the document]
Comment (fischer-ncar) [reply test/system/test_driver.sh:324]: Okay, I got it.  I pushed commits to use $wallclock_limit for derecho, cheyenne, and casper.

**Review by nusbaume — APPROVED** [review]
Thanks @fischer-ncar!  Everything looks good to me now.

@cacraigucar just a reminder to bring those changes into this PR before merging.  Thanks!

**Comment by cacraigucar** [comment]
This PR is being abandoned due to answer changes updating the externals.  We do not want to combine code changes with changes to externals which are answer changing.  Some of the items are addressed by PR #888

**Pull request closed without merging**